rover_auxiliary/src/ipcamera_api.py

```diff
@@ -0,0 +1,322 @@
+import logging, requests, signal, string
+from onvif import ONVIFCamera
+
+class TimeoutException(Exception):
+    pass
+
+class CameraController:
+    ONVIF_PARAMS = {
+        'Brightness': 'imaging',
+        'Contrast': 'imaging',
+        'ColorSaturation': 'imaging', 
+        'Sharpness': 'imaging',
+        'Resolution': 'video',
+        'FrameRateLimit': 'video',
+        'BitrateLimit': 'video'
+    }
+
+    HTTP_PARAMS = {
+        'WideDynamicRange': ('WDR', 1),
+        'WideDynamicRangeLevel': ('WDRLEVEL', 16),
+        'BackLight': ('BLC', 1),
+        'MirrorHorizontal': ('MHR', 1),
+        'MirrorVertical': ('MVR', 1),
+        'AntiFalseColor': ('AFC', 9),
+        'DigitalImageStabilizer': ('DIS', 12),
+        'LensShadeCorrection': ('LSC', 11),
+        'LensDistortionCorrection': ('LDC', 15),
+        'LensDistortionCorrectionLevel': ('LDCLevel', 19),
+        'AntiFog': ('AFG', 10),
+        'AntiFogLevel': ('AFGLevel', 20),
+        'SceneSelect': ('scene_Select', 2),
+        'ExposureMode': ('AEMode_Select', 2),
+        'ShutterSpeed': ('ShutterSpeedText', 2),
+        'ManualACG': ('AEGainText', 2),
+        'IRenable': ('IRenable', 0),
+        'IRmode': ('IRmode', 0),
+        'WhiteBalanceModeSelect': ('WBMode_Select', 3)
+    }
+
+    RESOLUTION_MAP = {
+        '480p': (640, 480),
+        'w480p': (720, 480),
+        '576p': (720, 576),
+        '720p': (1280, 720),
+        '960p': (1280, 960),
+        '1080p': (1920, 1080),
+        '1536p': (2048, 1536)
+    }
+
+    SHUTTER_MAP = {
+        "1/8000": 0,
+        "1/6000": 1,
+        "1/4000": 2,
+        "1/2000": 3,
+        "1/1000": 4,
+        "1/500": 5,
+        "1/250": 6,
+        "1/200": 7,
+        "1/150": 8,
+        "1/100": 9,
+        "1/50": 10,
+        "1/25": 11,
+        "1/20": 12,
+        "1/15": 13,
+        "1/10": 14,
+        "1/8": 15,
+        "1/5": 16,
+        "1/3": 17,
+        "1/2": 18,
+        "1": 19
+    }
+
+    AEGAIN_MAP = {
+        "1X": 0,
+        "2X": 1,
+        "4X": 2,
+        "8X": 3,
+        "16X": 4,
+        "32X": 5,
+        "64X": 6
+    }
+
+    def __init__(self, ip: str, port: int, username: str, password: str, timeout: int = 5):
+        self.ip = ip
+        self.port = port
+        self.username = username
+        self.password = password
+
+        # Configure logging
+        for handler in logging.root.handlers[:]:
+            logging.root.removeHandler(handler)
+        logging.basicConfig(level=logging.INFO, format='%(levelname)s - %(message)s')
+        self.logger = logging.getLogger()
+
+        def timeout_handler(signum, frame):
```

Comment:
Unless there's a specific use, I would define all function in the class at the same level (not inlined in the constructor). a lambda in the signal call would also work